defi/src/adaptors/handlers/storeAdaptorData/storeAll.ts

```diff
@@ -29,16 +31,83 @@ async function run() {
         todayIdSet = new Set(todayData.map((d: any) => d.id));
+        
+        // Load adaptor data to check dependencies and versions
+        const dataModule = loadAdaptorsData(adapterType);
+        const { protocolAdaptors, importModule } = dataModule;
+        
+        // Smart filtering: Build yesterdayIdSet by checking each protocol (similar to handler2 logic)
+        const now = new Date();
+        const currentHour = now.getUTCHours();
+        const startOfTodayTimestamp = getTimestampAtStartOfDayUTC(Math.floor(Date.now() / 1000)); // 00:00 UTC today
+        const isAfter1AM = currentHour >= 1;
+        const isAfter8AM = currentHour >= 8;
+        
+        // Process each protocol to determine if it should be in yesterdayIdSet
+        for (const protocol of protocolAdaptors) {
+          const id2 = protocol.id2;
+          const yesterdayRecord = yesterdayDataMap.get(id2);
+          
+          // If no yesterday data exists, don't add to set (will trigger refill)
+          if (!yesterdayRecord) {
+            continue;
+          }
+          
+          let includeInSet = true;
+          
+          try {
+            const adaptor = await importModule(protocol.module);
+            const version = adaptor.version ?? 1;
+            const runAtCurrTime = adaptor.runAtCurrTime ?? false;
+            const hasDune = adaptor.dependencies?.includes('dune' as any) ?? false;
+            const isV1 = version === 1;
+            const isV2 = !isV1;
+
+            // Edge case 1: V2 adapters (not runAtCurrTime) with incomplete yesterday data
+            // Only applies to V2 adapters that don't run at current time
+            // Remove from set to trigger refill for incomplete data after 01:00 UTC
+            if (isV2 && !runAtCurrTime && yesterdayRecord.updatedAt && yesterdayRecord.updatedAt < startOfTodayTimestamp && isAfter1AM) {
+              includeInSet = false;
+              console.log(`Removing ${id2} from yesterdayIdSet - incomplete V2 data (last update: ${new Date(yesterdayRecord.updatedAt * 1000).toISOString()})`)
+            }
+
+            // Edge case 2: V1 DUNE adapters with data updated before 08:00 UTC
+            // Only applies to V1 adapters with DUNE dependencies
+            // Remove from set to trigger refill after 08:00 UTC
+            if (includeInSet && isV1 && hasDune && yesterdayRecord.updatedAt && isAfter8AM) {
+              const lastUpdateDate = new Date(yesterdayRecord.updatedAt * 1000);
+              const lastUpdateHourUTC = lastUpdateDate.getUTCHours();
+
+              if (lastUpdateHourUTC < 8) {
+                includeInSet = false;
+                console.log(`Removing ${id2} from yesterdayIdSet - V1 DUNE adapter needs refresh (last update: ${lastUpdateDate.toISOString()}, before 08:00 UTC)`)
+              }
+            }
+          } catch (e) {
+            // If we can't load module, include by default to avoid breaking existing adapters
```

Comment:
can you add error logging here, so we could look into it if code ever reaches here